fuzz/fuzz_targets/utils.rs

```diff
@@ -1,71 +1,48 @@
+
+pub fn disable_unsupported_config(config: &mut wasm_smith::Config) {
+    config.gc_enabled = false;
+    config.exceptions_enabled = false;
+    config.max_memories = 1;
+    config.memory64_enabled = false;
+    config.relaxed_simd_enabled = false;
+    config.simd_enabled = false;
+    config.threads_enabled = false;
 }
 
-pub fn arbitrary_exec_module(seed: &[u8]) -> arbitrary::Result<wasm_smith::Module> {
-    let mut unstructured = Unstructured::new(seed);
-    wasm_smith::Module::new(exec_config(), &mut unstructured)
+pub fn arbitrary_config(unstructured: &mut Unstructured) -> arbitrary::Result<wasm_smith::Config> {
+    let mut config = wasm_smith::Config::arbitrary(unstructured)?;
+    disable_unsupported_config(&mut config);
+    Ok(config)
 }
 
-pub fn arbitrary_translate_module(seed: &[u8]) -> arbitrary::Result<wasm_smith::Module> {
-    let mut unstructured = Unstructured::new(seed);
-
-    let config = wasm_smith::Config::arbitrary(&mut unstructured);
-
-    config.map(|mut config| {
-        config.gc_enabled = false;
-        config.exceptions_enabled = false;
-        config.simd_enabled = false;
-        config.threads_enabled = false;
-
-        wasm_smith::Module::new(config, &mut unstructured)
-    })?
+pub fn arbitrary_swarm_config_module(
+    unstructured: &mut Unstructured,
+) -> arbitrary::Result<wasm_smith::Module> {
+    wasm_smith::Module::new(arbitrary_config(unstructured)?, unstructured)
 }
 
-/// Converts a [`ValType`] into a [`Val`] with default initialization of 1.
-///
-/// # ToDo
-///
-/// We actually want the bytes buffer given by the `Arbitrary` crate to influence
-/// the values chosen for the resulting [`Val`]. Also we ideally want to produce
-/// zeroed, positive, negative and NaN values for their respective types.
-pub fn ty_to_val(ty: &ValType) -> Val {
+/// Converts a [`ValType`] into an arbitrary [`Val`]
+pub fn ty_to_arbitrary_val(ty: &ValType, u: &mut Unstructured) -> Val {
     match ty {
-        ValType::I32 => Val::I32(1),
-        ValType::I64 => Val::I64(1),
-        ValType::F32 => Val::F32(1.0.into()),
-        ValType::F64 => Val::F64(1.0.into()),
-        unsupported => panic!(
-            "execution fuzzing does not support reference types, yet but found: {unsupported:?}"
+        ValType::I32 => Val::I32(u.int_in_range::<i32>(i32::MIN..=i32::MAX).unwrap_or(1)),
+        ValType::I64 => Val::I64(u.int_in_range::<i64>(i64::MIN..=i64::MAX).unwrap_or(1)),
+        ValType::F32 => Val::F32(
+            u.int_in_range::<u32>(u32::MIN..=u32::MAX)
+                .map(f32::from_bits)
+                .unwrap_or(1.0)
+                .into(),
+        ),
+        ValType::F64 => Val::F64(
+            u.int_in_range::<u64>(u64::MIN..=u64::MAX)
+                .map(f64::from_bits)
+                .unwrap_or(1.0)
+                .into(),
         ),
+        ValType::FuncRef => Val::FuncRef(FuncRef::null()),
+        ValType::ExternRef => Val::ExternRef(ExternRef::null()),
```

Comment:
Generally this is way nicer than before with the constant magic value. Though I am a bit astounded that there are no convenience functions to create simple Rust primitives (e.g. `i32`) from `&mut Unstructured`.

Hmmm, I think this is it: https://docs.rs/arbitrary/1.3.2/arbitrary/trait.Arbitrary.html#impl-Arbitrary%3C'a%3E-for-i32
`Arbitrary` impl for Rust primitives should do the job, right?